docs/enhancements/0002_day0-integration.md

```diff
@@ -0,0 +1,285 @@
+cluster, therefore, there is no much KMM can do to help deploy those kmods but
+it can manage their lifecycle once a customer has installed a cluster with his
+day-0 kmods in it - this is what this enhancement is about.
+
+### Day-0 cluster background
+
+A customer wishing to deploy OCP with custom nodes will need to generate 2
+artifacts using [image-builder](https://www.redhat.com/en/topics/linux/what-is-an-image-builder)
+  * A custom ISO for the nodes
+  * A container image encapsulating the ostree-commit that is deployed in the ISO
+
+The ISO will be used by [assisted-installer](https://docs.openshift.com/container-platform/4.14/installing/installing_on_prem_assisted/installing-on-prem-assisted.html)
+to install the cluster and the container image will be used to notify
+[MCO](https://docs.openshift.com/container-platform/4.14/post_installation_configuration/machine-configuration-tasks.html#understanding-the-machine-config-operator)
+that the image on the nodes is a custom image.
+
+When a customers has installed a "day-0 cluster" as mentioned above, we can
+assume that there is a `MachineConfig` in the cluster that looks something like
+```
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  creationTimestamp: "2023-12-31T14:03:08Z"
+  generation: 1
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 99-ybettan-external-image
+  resourceVersion: "1758"
+  uid: d1f07169-4cb5-4b85-93b3-3cd80530f028
+spec:
+  osImageURL: quay.io/ybettan/rhcos:414.92.202312311229-0
+```
+The important piece in it is `osImageURL: quay.io/ybettan/rhcos:414.92.202312311229-0`
+which tells MCO what that the ostree-commit that is supposed to be running on
+the node is encapsulated in `quay.io/ybettan/rhcos:414.92.202312311229-0`.
+
+If MCO, finds out that the OS on the node is different than the image specified
+in `osImageURL` then MCO will override the node OS based on the `MachineConfig`
+and reboot - we are going to leverage this mechanisem for the integration with
+KMM.
+
+In a day-0-cluster scenario, the user has applied this `MachineConfig` to tell
+MCO he has modified the OS on the nodes and that it shoudn't override it with
+the RHCOS image from the payload.
+
+### Setting up the integration
+
+A user that wishes to hand over the management of his day-0 kmod to KMM will
+first need to make sure the `MachineConfig` can be reconize by KMM, therefore,
+he would add a `machineconfiguration.openshift.io/kmm-managed: "true"` label to
+it.
+
+KMM will watch for `MachineConfig`s in the cluster containing the
+`machineconfiguration.openshift.io/kmm-managed: "true"` label and create a new
+`ModuleDay0` CR in the cluster for each one of them.
+We will have a different controller for this new CRD in KMM.
+**There should be no more than 1 `ModuleDay0` targeting a `MachineConfigPool`
+(masters/workers).
+If multiple kmods are required they should all be backed in the same image**
```

Comment:
What happens today if two `MachineConfig` objects setting different `osImageURL` values target the same `MachineConfigPool`?

Reply:
Never tries that. I would expect MCO to set an error in the `MachinecConfig`s statuses and not process any of them.

Reply:
backed -> baked

Reply:
> backed -> baked

Fixed. Thanks